roiextractors/extractors/schnitzerextractor/extractsegmentationextractor.py

```diff
@@ -149,3 +149,80 @@ def get_roi_ids(self):
 
     def get_image_size(self):
         return self._image_masks.shape[0:2]
+
+
+class ExtractPublicSegmentationExtractor(SegmentationExtractor):
+    """
+    This class inherits from the SegmentationExtractor class, having all
+    its funtionality specifically applied to the dataset output from
+    the \'EXTRACT\' ROI segmentation method.
+    """
+
+    extractor_name = "ExtractPublicSegmentationExtractor"
+    installed = HAVE_H5PY  # check at class level if installed or not
+    is_writable = False
+    mode = "file"
+    installation_mesg = "To use extract install h5py: \n\n pip install h5py \n\n"  # error message when not installed
+
+    def __init__(self, file_path: PathType):
+        """
+        Parameters
+        ----------
+        file_path: str
+            The location of the folder containing dataset.mat file.
+        """
+        SegmentationExtractor.__init__(self)
+        self.file_path = file_path
+        self._dataset_file = self._file_extractor_read()
+        self._image_masks = self._image_mask_extractor_read()
+        extraced_traces = self._trace_extractor_read()
+        if self._dataset_file['config']['preprocess']: #TODO: check for 'skip_dff' fields also
+            self._roi_response_raw = extraced_traces
+        else:
+            self._roi_response_dff = extraced_traces
+        self._sampling_frequency = (
+            extraced_traces.shape[1] / self._tot_exptime_extractor_read()
+        )
+        self._image_correlation = self._summary_image_read()
+
+    def __del__(self):
+        self._dataset_file.close()
+
+    def _file_extractor_read(self):
+        return h5py.File(self.file_path, "r")
+
+    def _image_mask_extractor_read(self):
+        return DatasetView(
+            self._dataset_file["spatial_weights"]
+        )
+
+    def _trace_extractor_read(self):
+        return DatasetView(
+            self._dataset_file["temporal_weights"]
+        ).lazy_transpose()
+
+    def _tot_exptime_extractor_read(self):
+        return self._dataset_file["info"]["time"]["totalTime"][0][0]#TODO
+
+    def _summary_image_read(self):
+        summary_image = self._dataset_file["info"]["summary_image"]#TODO
+        return np.array(summary_image)
+
+    def get_accepted_list(self):
+        return list(range(self.get_num_rois()))
+
+    def get_rejected_list(self):
+        ac_set = set(self.get_accepted_list())
+        return [a for a in range(self.get_num_rois()) if a not in ac_set]
+
+    @staticmethod
+    def write_segmentation(
+            segmentation_object: SegmentationExtractor, save_path, overwrite=True
+    ):
+        raise NotImplementedError
+
+    def get_roi_ids(self):
+        return list(range(self.get_num_rois()))
```

Comment:
Maybe this should be in the `SegmentationExtractor` class as a default implementation?